core/source/Number.mint

```diff
@@ -42,9 +42,13 @@ module Number {
   fun fromString (input : String) : Maybe(Number) {
     `
     (() => {
-      let value = parseFloat(#{input})
+      let value = Number(#{input})
```

Comment:
This seems like a breaking change (from this https://stackoverflow.com/questions/11988547/what-is-the-difference-between-number-and-parsefloat):
```
parseFloat('3.23abc'); //=> 3.23
Number('3.23abc'); //=> NaN
```

Not sure which one is better though :man_shrugging: maybe try parsing `Number` first and if it fails fallback to `parseFloat`? @Sija any opinions?

Reply:
Changing this was actually the main motivation behind this PR. Without this reduction, `Validate.isNumber()` will validate when a zip input has the value `1234x`. Someone may accidentally mistype his age as `8p`. It gets accepted and suddenly he is 8 instead of 80.

Reply:
I think we need to split this function into two with appropriate documentation:

1. `Number.fromString` should be strict, using `Number` and return `Maybe::Nothing` instead of `0` for empty string
	
2. `Number.fromStringRelaxed` should be relaxed, using `parseFloat`

Also, possibly return `Maybe::Nothing` if someone passes a non `String` value (in case of inline JS) for both.

Another option would be to have a second argument with a default value but I don't like that much.

Reply:
All good points IMO, I'd go for `Number.fromString(string, strict: true)` signature though, since it's essentially the same behaviour, just varied by the amount of validation.

Reply:
> All good points IMO, I'd go for Number.fromString(string, strict: true) signature though, since it's essentially the same behaviour, just varied by the amount of validation.

It is a pity, though, that we can not could call it like this: `Number.fromString("1", strict=true)`. 

> TIL: nifty trick to do a strict number (float) conversion

It can be done [even shorter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus): `+"x-1.11"` ;-)

Reply:
> It is a pity, though, that we can not could call it like this: `Number.fromString("1", strict=true)`.

I'm planning to implement something like that 😉 (would be `Number.fromString("1", strict: true)` to match other things).

We can go with the second argument if you both think it's better :+1:

Reply:
I will do.

Reply:
- I'd say it should be added to the language, perhaps as a const (`Number.Infinity`)?
- Good point
- I'd say it's either this, or error

Reply:
* Yes, a constant for it makes sense.
* That will need to change to return `Result(Number.Error, String)`
* It's consistent with JavaScript so I guess it can stay as is.
* `Number.toFixed` needs to return a result as well since it can throw a `RangeError` `Infinity.toFixed(200)`

Reply:
Ok, but it is a lot of work, basically, go through the whole library and check, and the point of this PR was not to add Infinity to Mint (It did not exist before), but to fix the cases when `Validate.isNumber()` shows a weird behavior.

Further, If the addition of Infinity leads to code being more bloated, like, `Number.format` returns `Result(Number.Error, String)` instead of just `String`, then this is a big problem in my opinion. Give me some practical use cases for Infinity, that justify this overhead for library users.

Reply:
Ok, I just see, parseFloat(Infinity) actually returns Infinity. So this was an unhandled edge case before.

Reply:
I will handle it in this function, but keep the other functions as they were for now.

Reply:
> Give me some practical use cases for Infinity, that justify this overhead for library users.

I can't really :slightly_smiling_face: but it could happen to somebody (that's why for example `toFixed` raises an error on big `digits` argument) but until it does, we can leave the things as is.